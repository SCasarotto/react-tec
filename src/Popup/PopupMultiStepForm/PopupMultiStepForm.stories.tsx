import React from 'react';

import { Story, Meta } from '@storybook/react';

import { Label } from '../../Form/Label';
import {
  PopupMultiStepForm,
  PopupMultiStepFormProps,
} from './PopupMultiStepForm';

export default {
  title: 'Popup/PopupMultiStepForm',
  component: PopupMultiStepForm,
  argTypes: {},
} as Meta;

const Template: Story<PopupMultiStepFormProps> = (args) => (
  <PopupMultiStepForm {...args} />
);

export const Default = Template.bind({});
Default.args = {
  visible: true,
  onClose: () => console.log('onClose'),
  onSubmit: async () => console.log('MultiStepForm Submit'),
  stepData: [
    {
      title: 'Step210',
      component: (
        <div>
          <Label htmlFor='test' required disabled>
            Label
          </Label>
        </div>
      ),
    },
    {
      title: 'Step#3',
      component: (
        <div>
          <Label htmlFor='test' required disabled>
            S33
          </Label>
        </div>
      ),
    },
    {
      title: 'Step444',
      component: (
        <div>
          <Label htmlFor='test' required disabled>
            A4A4
          </Label>
        </div>
      ),
    },
  ],
};
