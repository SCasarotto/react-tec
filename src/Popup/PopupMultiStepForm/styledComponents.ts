import styled from 'styled-components';

import { Popup as P } from '../Popup';

export const Popup = styled(P)`
  .rec-popupContent {
    width: 90%;
    max-width: 900px;
    margin: 100px auto;
    padding: 30px;
  }
`;
