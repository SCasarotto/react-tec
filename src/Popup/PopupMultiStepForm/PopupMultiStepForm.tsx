import React from 'react';

import { MultiStepForm, MultiStepFormProps } from '../../Form/MultiStepForm';
import { PopupProps } from '../Popup';
import { Popup } from './styledComponents';

export interface PopupMultiStepFormProps
  extends PopupProps,
    MultiStepFormProps {
  onClose(): void;
  onSubmit(): Promise<any>;
}
/**
 *
 * PopupMultiStepForm is a combination of `Popup` and `MultiStepForm`. Besides the props below, reference `MultiStepForm` for additional props.
 *
 */
export const PopupMultiStepForm = React.forwardRef<
  HTMLFormElement,
  PopupMultiStepFormProps
>((props, ref) => {
  const { visible, onClose, className = '', ...rest } = props;

  return (
    <Popup visible={visible} className={`tec-popupMultiStepForm ${className}`}>
      <MultiStepForm
        handleCancelOnClick={onClose}
        className='tec-popupMultiStepFormMultiStepForm'
        ref={ref}
        {...rest}
      />
    </Popup>
  );
});

PopupMultiStepForm.defaultProps = {
  visible: false,
};
