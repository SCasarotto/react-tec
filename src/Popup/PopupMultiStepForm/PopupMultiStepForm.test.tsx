import React from 'react';

import { render, waitFor } from '@testing-library/react';

import { PopupMultiStepForm } from './PopupMultiStepForm';

describe('PopupMultiStepForm', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(
      <PopupMultiStepForm
        visible
        stepData={[]}
        onClose={jest.fn()}
        onSubmit={async () => jest.fn()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(
      <PopupMultiStepForm
        visible
        stepData={[]}
        onClose={jest.fn()}
        onSubmit={async () => jest.fn()}
        className='myClassName'
      />,
    );
    expect(container.firstChild).toHaveClass('myClassName');
  });

  // Ref
  it('renders with ref', async () => {
    const ref: { current: HTMLFormElement | null } = { current: null };
    render(
      <PopupMultiStepForm
        visible
        stepData={[]}
        onClose={jest.fn()}
        onSubmit={async () => jest.fn()}
        ref={ref}
        data-testid='myFormId'
      />,
    );
    // Using this as the focus check didn't seem to work
    waitFor(() => expect(ref.current).toBeTruthy());
  });
});
