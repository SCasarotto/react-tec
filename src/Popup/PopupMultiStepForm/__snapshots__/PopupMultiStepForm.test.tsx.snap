// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupMultiStepForm matches snapshot 1`] = `
<DocumentFragment>
  .c5 {
  position: relative;
  width: 100%;
  margin: 0px auto;
  box-sizing: border-box;
}

.c7 {
  position: relative;
  display: block;
  width: 100%;
  font-size: 18px;
  padding: 5px 20px;
  text-align: center;
  cursor: pointer;
  border-radius: 5px;
  box-sizing: border-box;
  margin: 0px;
  border: 1px solid;
  -webkit-transition: color 0.2s ease-in,background-color 0.2s ease-in, border 0.2s ease-in;
  transition: color 0.2s ease-in,background-color 0.2s ease-in, border 0.2s ease-in;
}

.c7:hover,
.c7:active {
  border: 1px solid;
}

.c7:disabled {
  cursor: not-allowed;
  border: 1px solid;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin-bottom: 30px;
}

.c4 {
  position: absolute;
  top: 14px;
  height: 2px;
  width: -Infinity%;
  left: Infinity%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  position: fixed;
  width: 100%;
  height: 100%;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.5);
  z-index: 1000;
  overflow-y: auto;
  -webkit-transition: visibility 0.25s ease-in-out,opacity 0.25s ease-in-out;
  transition: visibility 0.25s ease-in-out,opacity 0.25s ease-in-out;
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: relative;
  border-radius: 20px;
  margin: 50px auto;
  max-width: 650px;
}

.c1 .PopupContent {
  width: 90%;
  max-width: 900px;
  margin: 100px auto;
  padding: 30px;
}

@media (max-width:500px) {
  .c3 {
    margin-bottom: 15;
  }
}

<div
    class="c0 tec-popup c1 tec-popupMultiStepForm "
  >
    <div
      class="c2 tec-popupContent"
    >
      <div
        class="tec-multiStepForm tec-popupMultiStepFormMultiStepForm"
      >
        <div
          class="c3 tec-multiStepFormStepContainer"
        >
          <div
            class="c4 tec-multiStepFormStepBar"
          />
        </div>
        <form
          class="c5 tec-form tec-multiStepFormForm"
        >
          <div
            class="c6 tec-multiStepFormButtonContainer"
          >
            <button
              class="c7 tec-button tec-multiStepFormStepButton tec-multiStepFormStepButtonLeft"
              type="button"
            >
              cancel
            </button>
            <button
              class="c7 tec-button tec-multiStepFormStepButton tec-multiStepFormStepButtonright"
              type="button"
            >
              next
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
